array-reduce/main.js

```diff
@@ -0,0 +1,55 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+const sum = account.reduce((acc, currValue) => {
+  return acc += currValue.amount
+}, 0)
+console.log('sum:', sum);
+
+const product = account.reduce((acc, currValue) => {
+  return acc *= currValue.amount
+}, 1)
+console.log('product:', product);
+
+const deposit = account.reduce((acc, curr) => {
+  if (curr.type === 'deposit') {
+    acc += curr.amount
+  }
+  return acc;
+}, 0)
+
+const withdrawal = account.reduce((acc, curr) => {
+  if (curr.type === 'withdrawal') {
+    acc += curr.amount
+  }
+  return acc;
+}, 0)
+
+const balance = deposit - withdrawal;
+console.log('balance:', balance)
```

Comment:
This can be done with a single `reduce` call. Here is some starter code to point you in the right direction:

```JAVASCRIPT
const balance = account.reduce((total, transaction) => {
  // If the transaction is a deposit add the "amount" to the "total"
  
  // if the transaction is a withdrawal subtract the "amount" from the "total"
  
  // return the "total"
}, 0)
```